designs/capi-resources-custom-namespace.md

```diff
@@ -0,0 +1,54 @@
+# CAPI Resources in Custom Namespace
+
+## Introduction
+
+**Problem:** Customers managing large scale EKS-A clusters (hundreds or thousands of workload clusters for a given management cluster) expect to have difficulty managing all these resources if they are all in the same namespace 
+
+This document aims to solve that, by allowing customers to specify a new flag on the cluster spec. If set, the underlying CAPI/CAPX/etcd resources will be created under the Cluster's namespace, rather than all dumped into `eksa-system`. 
+This change would make it so customers would be able to put these resources virtually in any namespace, despite the fact that they may be considered eks-a internal components.
```

Comment:
should we prevent customers from using certain namespaces? For example, `kube-system` and `default` come to mind. `default` can be subject to accidental exposure and `kube-system` exposes a path from our components to critical system components that would be better avoided.

Reply:
Good call! We can definitely add a denylist for these namespaces and validate that on cluster creation.